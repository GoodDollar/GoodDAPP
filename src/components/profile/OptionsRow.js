// @flow
import React from 'react'
import { View } from 'react-native'
import { RadioButton } from 'react-native-paper'
import { startCase } from 'lodash'
import { withStyles } from '../../lib/styles'
import { Text } from '../common'

// privacy options
const privacyOptions = ['private', 'masked', 'public']

/**
 * OptionsRow component
 * @param {object} props
 * @param {string} props.title
 * @param {object} props.styles
 * @returns {React.Node}
 * @constructor
 */
const OptionsRow = ({ title = '', styles, theme }) => (
  <View style={styles.optionsRowContainer}>
    <Text style={styles.growTwo} textAlign="left" color="gray" fontWeight="medium">
      {title}
    </Text>

    {privacyOptions.map(privacy => (
      <View style={styles.optionsRowTitle} key={privacy}>
        {title === '' ? (
          <Text fontSize={14} color="gray">
            {startCase(privacy)}
          </Text>
        ) : (
          <RadioButton value={privacy} uncheckedColor={theme.colors.gray} color={theme.colors.primary} />
        )}
      </View>
    ))}
  </View>
)

const getStylesFromProps = ({ theme }) => {
  return {
    optionsRowContainer: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      borderStyle: 'solid',
      borderBottomColor: theme.colors.lightGray,
      borderBottomWidth: 1,
      padding: theme.paddings.mainContainerPadding,
      paddingLeft: theme.sizes.defaultQuadruple,
    },
    growTwo: {
      flexGrow: 2,
    },
    optionsRowTitle: {
      width: '15%',
      minWidth: 60,
      alignItems: 'center',
    },
  }
}

export default withStyles(getStylesFromProps)(OptionsRow)
