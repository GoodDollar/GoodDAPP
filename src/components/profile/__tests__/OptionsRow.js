import React from 'react'
import renderer from 'react-test-renderer'
import { withThemeProvider } from '../../../__tests__/__util__'
import OptionsRow from '../OptionsRow'

describe('OptionsRow', () => {
  const WrappedOptionsRow = withThemeProvider(OptionsRow)

  it('matches snapshot', async () => {
    let component

    // eslint-disable-next-line require-await
    await renderer.act(async () => (component = renderer.create(<WrappedOptionsRow />)))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
