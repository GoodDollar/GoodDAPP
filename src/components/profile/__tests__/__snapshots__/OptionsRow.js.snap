// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionsRow matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-borderBottomColor-1fn3lry r-borderBottomWidth-qklmqi r-borderStyle-1phboty r-display-6koalj r-flexDirection-18u37iz r-padding-edyy15 r-paddingLeft-gu0qjt"
  >
    <div
      className="css-text-901oao r-color-181bzza r-flexGrow-16xksha r-fontFamily-vnw8o6 r-fontSize-ubezar r-fontWeight-majxgm r-lineHeight-qvk6io r-textAlign-fdjqy7 r-textDecorationLine-13wfysu r-textTransform-3twk1y"
      dir="auto"
    >
      
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-minWidth-1c8i11 r-width-1ih7r1r"
    >
      <div
        className="css-text-901oao r-color-181bzza r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        Private
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-minWidth-1c8i11 r-width-1ih7r1r"
    >
      <div
        className="css-text-901oao r-color-181bzza r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        Masked
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-minWidth-1c8i11 r-width-1ih7r1r"
    >
      <div
        className="css-text-901oao r-color-181bzza r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        Public
      </div>
    </div>
  </div>
</div>
`;
